### PR TITLE
Allow browsing loadouts using hotkeys

### LINKED ISSUE #41: Select next/prev loadout
Need a global keybind for selecting next/previous loadout.

Got the wrong loadout selected on drop ?
Going into an eradicate mission ? 
Alt+tab slow ? 

Quick switch to the correct loadout, using the new speedkeys!

Make +/- slect the next or previous loadout, looping around when on last/first item.
Make the keys rebindable in settings.

### PR DESCRIPTION
**Pull request by stixes**
Added hotkeys for cycling through loadouts up and down.

* Ensured hotkeys are disabled when armed
* Keys are configurable from settings
* defaults are present in Settings
* Lightly refactored input handling in PynputKeyListener

Fixes #41 

Known problems: Crash when attempting to reassign key. This problem is also present on main

### CODE REVIEW

File: src/controller.py [hunk 1]
```diff
@@ -48,6 +48,42 @@ def set_armed(self, isArmed):
...
+        current_index = loadout_ids.index(current_loadout_id)
+        next_index = (current_index + 1) % len(loadout_ids)
+        
+        # Set the next loadout as active
+        self.model.set_active_loadout(loadout_ids[next_index])
+        
+        # Debug information
+        print(f"DEBUG: New loadout selected: {self.model.currentLoadout.name}")
+        
+        # FIXME: Issue with updating the view due to threading
```
Comment (stixes) [inline src/controller.py:66]: @Noloxs Er det noget du ved hvordan man kan fikse?

Comment (Noloxs) [reply src/controller.py:66]: Jeps.
Men det vil kræver noget re-work af view håndteringen.

Kan lige se på det senere.

File: src/controller.py [hunk 2]
```diff
@@ -48,6 +48,36 @@ def set_armed(self, isArmed):
...
+        # Calculate the index of the next loadout
+        current_index = loadout_ids.index(current_loadout_id)
+        prev_index = (current_index - 1 + len(loadout_ids)) % len(loadout_ids)
+        
+        # Set the next loadout as active
+        self.set_active_loadout(loadout_ids[prev_index])
+        
+        # Debug information
+        print(f"DEBUG: New loadout selected: {self.model.currentLoadout.name}")
+
```
Comment (Noloxs) [inline src/controller.py:80]: Should be extracted into a method that can handle back and forth

File: src/listener_pynput.py [hunk 3]
```diff
@@ -10,7 +10,36 @@ def __init__(self, model, controller):
         self.listener = keyboard.Listener(on_press=self.on_press, on_release=self.on_release, suppress=False)
         self.listener.start()
         self.getNextCallbacks = []
+        self.key_press_handlers = {
+            self.globalArmKey: self.handle_global_arm_press,
+            # TODO: Put these keys in settings
+            "+": self.handle_next_loadout,
+            "-": self.handle_prev_loadout,
+        }
```
Comment (Noloxs) [inline src/listener_pynput.py:18]: Need to be added to settings, with a on/off option

**Comment by Noloxs** [comment]
Hot swapping loadouts should also be disabled if armed

File: src/listener_pynput.py (hunk 3 above)
Comment (stixes) [reply src/listener_pynput.py]: fixed

**Comment by Noloxs** [comment]
Merged main to branch and fixed issue with changing inputs

**Review by Noloxs — APPROVED** [review]

**Comment by Noloxs** [comment]
@stixes Merge if happy

**Review by Noloxs — CHANGES_REQUESTED** [review]
Found bug:

List of observers are added to settings.json when saving.
This makes it crash on start

**Comment by stixes** [comment]
> Found bug:
> 
> List of observers are added to settings.json when saving. This makes it crash on start
> 
> Fails test

Oops

**Review by Noloxs — APPROVED** [review]

**Pull request merged**